tests/periph_gpio_arduino/main.sketch

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (C) 2018 Tom Keddie
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Manual test application for GPIO peripheral drivers using
+ *              Arduino pin numbering
+ *
+ * @author      Tom Keddie <github@bronwenandtom.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "shell.h"
+#include "periph/gpio.h"
+
+#define ARDUINO_PIN_NUMOF ((int)(sizeof(arduino_pinmap)/sizeof(arduino_pinmap[0])))
+
+static int get_digital_pin(const char* pinstring)
+{
+    int arduino_pin = -1;
+    char *endptr  = NULL;
+
+    if (*pinstring == 'A') {
+        // TODO : add analogInputToDigitalPin macro to arduino_board.h for all boards
+        return -1;
+    }
+
+    arduino_pin = strtol(pinstring, &endptr, 0);
+
+    if ((*endptr != '\0') || (arduino_pin == -1) || (arduino_pin > ARDUINO_PIN_NUMOF)) {
```

Comment:
`strtol` returns `0` on conversion error, hence `-1` doesn't make sense here, right? See also http://www.cplusplus.com/reference/cstdlib/strtol/

Reply:
Hence you may (need to) default to pin 0, anyway?

Reply:
What an awful API, thanks for picking that up, I'll write my own.